client/core/bookie.go

```diff
@@ -1055,6 +1171,16 @@ func handleEpochReportMsg(_ *Core, dc *dexConnection, msg *msgjson.Message) erro
 	if err != nil {
 		return fmt.Errorf("epoch report note unmarshal error: %w", err)
 	}
+	// Acquiring dc.booksMtx here in dc.bookie serves as a barrier for notifications that
+	// come after we sent new subscription request (e.i. notes that need to be applied
+	// on top of snapshot we get from new subscription request will wait on that mutex
+	// here). Note, we don't acquire dc.booksMtx for the whole handleBookOrderMsg
+	// duration; that leaves some room for notifications from previous subscription to
+	// come and apply concurrently with us applying snapshot we just got from new
+	// subscription request.
+	// TODO: I'm not sure at the moment whether is OK to execute this notification from
+	//  old subscription while Resetting order book concurrently. Could this result into
+	//  mixed OrderBook state (for orders, candles cache, or something else) ?
```

Comment:
Would like to know answer to that TODO question (that's in `handleEpochReportMsg` **func**).

Reply:
I don't really see anything in `epoch_report` notification that would need to be atomic with respect to current `orders` in the book or `seq` number, so probably don't even need this "barrier" here (and locking `dc.booksMtx` in `dc.bookie` just to get the bookie).